logstash-core/lib/logstash/plugin.rb

```diff
@@ -122,12 +123,22 @@ def metric
                          LogStash::Instrument::NamespacedNullMetric.new(@metric, :null)
                        end
   end
+
   # return the configured name of this plugin
   # @return [String] The name of the plugin defined by `config_name`
   def config_name
     self.class.config_name
   end
 
+  # commit event to dlq
+  def dlq_commit(event, reason)
```

Comment:
I was working with someone the other day that wanted to dead-letter-queue events that could not be deserialized using the avro-codec upon input from kafka. So that would be an example of the input being able to DLQ.

That being said, we may not want to handle such a scenario, and I am cool with that!